### PR TITLE
feat: add cypress integration

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
This PR will:

- add cypress integration
- add create deal e2e test

## Checklist

- [x] I used the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
- [x] Unit tests have been created if a new feature was added.
- [ ] Translations have been added and extracted if a new feature was added.

### CODE REVIEW

File: cypress/integration/deals.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+
```
Comment (BouweCeunen) [inline cypress/integration/deals.ts]: small thing, redundant enter

File: cypress/support/index.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { RPCEndpoint } from "../../src/types/solana.types";
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace Cypress {
+    interface Chainable {
+      /**
+       * Custom command to select DOM element by data-cy attribute.
+       * @example cy.dataCy('greeting')
+       */
```
Comment (BouweCeunen) [inline cypress/support/index.ts]: maybe redundant information?

**Review by BouweCeunen — DISMISSED** [review]

**Review by BouweCeunen — DISMISSED** [review]

File: cypress/integration/deals.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+before(() => {
+	let wallet;
+
+	cy.visit("http://localhost:3000/");
+
+	cy.window()
+		.then((win) => {
+			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+			// @ts-ignore
+			wallet = win.solana;
```
Comment (RuvendeGroote) [inline cypress/integration/deals.ts:18]: what is this win.solana? can we explain it in a comment

**Review by RuvendeGroote — DISMISSED** [review]

File: cypress/integration/deals.ts (hunk 3 above)
Comment (BartDeCaluwe) [reply cypress/integration/deals.ts:18]: `win.solana` is a solana object injected by phantom. 
More info in the [docs](https://docs.phantom.app/integrating/extension-and-mobile-browser/detecting-the-provider) 

I'll add an excerpt from the docs as a comment.

**Review by RuvendeGroote — APPROVED** [review]

**Pull request merged**